wallet-core/src/profile.rs

```diff
@@ -12,14 +12,27 @@ use sha2::{Digest, Sha256};
 use execution_core::{
     signatures::bls::SecretKey as BlsSecretKey,
     transfer::phoenix::{
-        PublicKey as PhoenixPublicKey, SecretKey as PhoenixSecretKey,
+        Note, PublicKey as PhoenixPublicKey, SecretKey as PhoenixSecretKey,
         ViewKey as PhoenixViewKey,
     },
 };
 use zeroize::Zeroize;
 
 use crate::RNG_SEED;
 
+/// Calculate the sum for all the given [`Note`]s that belong to the given
+/// [`PhoenixViewKey`].
+///
+/// This function assumes that all notes are owned by the given view-key.
+pub fn phoenix_balance(
+    phoenix_vk: &PhoenixViewKey,
+    notes: impl AsRef<[Note]>,
+) -> u64 {
+    notes.as_ref().iter().fold(0, |acc, note| {
+        acc + note.value(Some(phoenix_vk)).unwrap_or_default()
+    })
+}
+
```

Comment:
This is not the function, neither the algorithm, we have at the moment.
First of all, it needs to returns a `BalanceInfo` (that contains both `value` and `spendable`), and in order to do so it needs to `sort` the notes before calculate the `spendable`.

Reply:
Was too fast and didn't look properly. Should be fixed now!